custom/icds_reports/static/js/directives/download/download.directive.js

```diff
@@ -385,8 +380,20 @@ function DownloadController($rootScope, $location, locationHierarchy, locationsS
             vm.onSelectYear({'id': vm.selectedYear});
             vm.selectedFormat = 'xlsx';
         }
+
+        vm.adjustSelectedLevelForNoViewByFilter();
```

Comment:
it might be simpler if `vm.selectedLevel` were set to a function that just calculated it on demand (since the definition is always the same, the issue was just it not being updated)

Reply:
yeah. That can be. I think this is used as a model in the HTML element. So don't wanna touch that right now as this bug is breaking the report so would prefer to merge it first.